### PR TITLE
feat: new solutions upgrade

### PR DESCRIPTION
**Pull request by surajsbharadwaj**
### Description

1. Downgrade terraform ibm provider version to 1.49.0 to make it compatible with secure landing zone vpc module.
2. Reorganized the example folder structure
3. Upgraded the examples to have a full stack solution (secure landing zone vpc plus Powervs infra deployed in one go).
4. Renamed the flavour for IBM catalog

#### No release required

- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [ ] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [ ] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [x] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content

1. Downgrade terraform ibm provider version to 1.49.0 to make it compatible with secure landing zone vpc module.
2. Reorganized the example folder structure
3. Upgraded the examples to have a full stack solution (secure landing zone vpc plus Powervs infra deployed in one go).
4. Renamed the flavour for IBM catalog
5. Added json validation for powervs extension flavour

---

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

**Comment by surajsbharadwaj** [comment]
@kbiegert updated the ibm catalog json file.
@Beilmann : need your help for the description (under image in ibm_ctalog.json) and in the readmes.
@SirSpidey need your help for review.

**Review by kbiegert — APPROVED** [review]
I reviewed the ibm_catalog.json changes and they look reasonable to me.  Once the catalog delivers support for "is_fullstack" you will able to name the variations the same name.

File: reference-architectures/extension/deploy-arch-ibm-pvs-inf-extension.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
+---
```
Comment (SirSpidey) [inline reference-architectures/extension/deploy-arch-ibm-pvs-inf-extension.md:1]: I created a PR from this branch for the changes to the reference architecture.

File: reference-architectures/full-stack/deploy-arch-ibm-pvs-inf-full-stack.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ deployment-url: url #TODO
 
 docs: https://cloud.ibm.com/docs/solution-guide #TODO
 
-image_source: https://github.com/terraform-ibm-modules/terraform-ibm-powervs-infrastructure/reference-architectures/standard/deploy-arch-ibm-pvs-inf-standard.svg
+image_source: https://github.com/terraform-ibm-modules/terraform-ibm-powervs-infrastructure/reference-architectures/full-stack/deploy-arch-ibm-pvs-inf-full-stack.svg
```
Comment (SirSpidey) [inline reference-architectures/full-stack/deploy-arch-ibm-pvs-inf-full-stack.md:19]: updated this in a separate PR

**Review by SirSpidey — CHANGES_REQUESTED** [review]
Created https://github.com/terraform-ibm-modules/terraform-ibm-powervs-infrastructure/pull/358 to merge into this PR for my suggestions for the reference architectures.

Otherwise, small suggestions.

File: examples/terraform-registry/extension/README.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # Standard example for Power infrastructure for regulated industries
 
-This example illustrates how to use the `power-infrastructure` module in the IBM catalog. It provisions the following infrastructure:
+This example illustrates how to use the `power-infrastructure` module. It provisions the following infrastructure:
```
[the remark this answers is not in the document]
Comment (Beilmann) [reply examples/terraform-registry/extension/README.md:3]: @surajsbharadwaj - remove all occurances of "Power infrastructure for regulated industries" and replace with "Power infrastructure for deployable architectures"

File: ibm_catalog.json [hunk 4]
```diff
@@ -5,10 +5,201 @@
             "label": "Power infrastructure for deployable architectures",
             "flavors": [
                 {
-                    "label": "PowerVS workspace",
-                    "name": "standard",
-                    "working_directory": "examples/ibm-catalog/standard-solution",
-                    "usage_template": "module \"power-infrastructure\" {\n  source =                     \"https://cm.globalcatalog.cloud.ibm.com/api/v1-beta/offering/source//${{workingDirectory}}?archive=tgz&flavor=${{flavor}}&kind=${{kind}}&name=${{name}}&version=${{version}}\"\n  prerequisite_workspace_id   = var.prerequisite_workspace_id\n  powervs_zone                = var.powervs_zone\n  powervs_resource_group_name = var.powervs_resource_group_name\n  ssh_private_key             = var.ssh_private_key\n  ibmcloud_api_key            = var.ibmcloud_api_key\n}",
+                    "label": "PowerVS Workspace incl. Secure Landing Zone",
```
Comment (Beilmann) [inline ibm_catalog.json:8]: @kbiegert - name is considered to be temporary. please inform if we get a switch between full-stack and extension variations

File: ibm_catalog.json [hunk 5]
```diff
@@ -5,10 +5,201 @@
...
+                                    "url": "https://raw.githubusercontent.com/terraform-ibm-modules/terraform-ibm-powervs-infrastructure/main/reference-architectures/full-stack/deploy-arch-ibm-pvs-inf-full-stack.svg",
+                                    "type": "image/svg+xml"
+                                },
+                                "description": "Full stack solution which deploys Secure infrastructure on VPC for regulated industries and deploys  Power Virtual Server workspace and connects it with VPC services. Proxy service for public internet access from PowerVS workspace is configured. \n\nAdditional management components (NFS server, NTP forwarder, and DNS forwarder) may be installed on VPC optionally."
+                            }
+                        ]
+                    }
+                },
+                {
+                    "label": "PowerVS Workspace Extension",
```
Comment (Beilmann) [inline ibm_catalog.json:199]: @kbiegert - name is considered to be temporary. please inform if we get a switch between full-stack and extension variations

File: ibm_catalog.json (hunk 4 above)
Comment (Beilmann) [inline ibm_catalog.json:8]: "label": "PowerVS workspace (incl. Secure Landing Zone)",

File: ibm_catalog.json (hunk 5 above)
Comment (Beilmann) [inline ibm_catalog.json:199]: "label": "PowerVS Workspace (extension)",

File: ibm_catalog.json [hunk 6]
```diff
@@ -5,10 +5,201 @@
...
+                            }
+                        ],
+                        "diagrams": [
+                            {
+                                "diagram": {
+                                    "caption": "Power infrastructure for deployable architectures",
+                                    "url": "https://raw.githubusercontent.com/terraform-ibm-modules/terraform-ibm-powervs-infrastructure/main/reference-architectures/full-stack/deploy-arch-ibm-pvs-inf-full-stack.svg",
+                                    "type": "image/svg+xml"
+                                },
+                                "description": "Full stack solution which deploys Secure infrastructure on VPC for regulated industries and deploys  Power Virtual Server workspace and connects it with VPC services. Proxy service for public internet access from PowerVS workspace is configured. \n\nAdditional management components (NFS server, NTP forwarder, and DNS forwarder) may be installed on VPC optionally."
```
Comment (Beilmann) [inline ibm_catalog.json:193]: "description": "'Power Infrastructure for deployable architecture' in full-stack variation 'PowerVS workspace' creates in one single deployment Power Virtual Server workspace and connects it with VPC services. \n\nProxy service for public internet access from PowerVS workspace is configured. \n\nYou can optionally install more management components (such as NFS server, NTP forwarder, and DNS forwarder) on VPC."

File: reference-architectures/extension/deploy-arch-ibm-pvs-inf-extension.md [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+subcollection: deployable-reference-architectures
+
+authors:
+  - name: Arnold Beilmann
+
+version: v7.0.0
+
+deployment-url: url #TODO
+
+docs: https://cloud.ibm.com/docs/solution-guide #TODO
```
[the remark this answers is not in the document]
Comment (Beilmann) [reply reference-architectures/extension/deploy-arch-ibm-pvs-inf-extension.md:17]: @SirSpidey - I do not know what final URL will be (at the moment). Have only an internal one

File: ibm_catalog.json [hunk 8]
```diff
@@ -114,37 +305,36 @@
...
+                                    "url": "https://raw.githubusercontent.com/terraform-ibm-modules/terraform-ibm-powervs-infrastructure/main/reference-architectures/extension/deploy-arch-ibm-pvs-inf-extension.svg",
                                     "type": "image/svg+xml"
                                 },
-                                "description": "Building upon the VPC service that were previously created when you deployed Secure infrastructure on VPC for regulated industries, Power Infrastructure for deployable architectures creates a Power Virtual Server workspace and connects it with VPC services. Proxy service for public internet access from PowerVS workspace is configured. \n\nAdditional management components (NFS server, NTP forwarder, and DNS forwarder) may be installed on VPC optionally."
+                                "description": "Deploys  Power Virtual Server workspace and connects it with VPC services. Proxy service for public internet access from PowerVS workspace is configured. \n\nAdditional management components (NFS server, NTP forwarder, and DNS forwarder) may be installed on VPC optionally."
```
Comment (Beilmann) [inline ibm_catalog.json:337]: "description": "Building upon the VPC service that were previously created when you deployed 'Secure infrastructure on VPC for regulated industries' or 'Power Infrastructure for deployable architecture' in a full-stack variation 'PowerVS Workspace', extension variation of ''PowerVS Workspace' creates a Power Virtual Server workspace and connects it with VPC services. \n\nProxy service for public internet access from PowerVS workspace is configured. \n\nYou can optionally reconfigure management components on VPC (such as NFS server, NTP forwarder, and DNS forwarder)."

File: reference-architectures/extension/deploy-arch-ibm-pvs-inf-extension.md [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+{{site.data.keyword.attribute-definition-list}}
+
+# Power infrastructure for deployable architectures - variation 'PowerVS workspace'
+{: #deploy-arch-ibm-pvs-inf-standard}
+{: toc-content-type="reference-architecture"}
+{: toc-industry="Technology"}
+{: toc-use-case="ITServiceManagement"}
+{: toc-compliance="SAPCertified"}
+
+Building upon the VPC service that were previously created when you deployed 'Secure infrastructure on VPC for regulated industries' or 'Power Infrastructure for deployable architecture' in variation 'full-stack', 'Power Infrastructure for deployable architectures' in variation 'extension' creates a Power Virtual Server workspace and connects it with VPC services. Proxy service for public internet access from PowerVS workspace is configured.
```
Comment (Beilmann) [inline reference-architectures/extension/deploy-arch-ibm-pvs-inf-extension.md:48]: Building upon the VPC service that were previously created when you deployed 'Secure infrastructure on VPC for regulated industries' or 'Power Infrastructure for deployable architecture' in full-stack variation 'PowerVS workspace', extension variation 'PowerVS workspace' creates a Power Virtual Server workspace and connects it with VPC services. Proxy service for public internet access from PowerVS workspace is configured.

File: reference-architectures/extension/deploy-arch-ibm-pvs-inf-extension.md [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+
+You can optionally (re)configure management components on VPC (such as NFS server, NTP forwarder, and DNS forwarder).
```
Comment (Beilmann) [inline reference-architectures/extension/deploy-arch-ibm-pvs-inf-extension.md:50]: You can optionally reconfigure management components on VPC (such as NFS server, NTP forwarder, and DNS forwarder).

File: reference-architectures/full-stack/deploy-arch-ibm-pvs-inf-full-stack.md [hunk 11]
```diff
@@ -45,19 +45,19 @@ content-type: reference-architecture
 {: toc-use-case="ITServiceManagement"}
 {: toc-compliance="SAPCertified"}
 
-Building upon the VPC service that were previously created when you deployed Secure infrastructure on VPC for regulated industries, Power Infrastructure for deployable architectures creates a Power Virtual Server workspace and connects it with VPC services. Proxy service for public internet access from PowerVS workspace is configured.
+'Power Infrastructure for deployable architecture' in variation 'full-stack' creates in one single deployment Power Virtual Server workspace and connects it with VPC services. Proxy service for public internet access from PowerVS workspace is configured.
```
Comment (Beilmann) [inline reference-architectures/full-stack/deploy-arch-ibm-pvs-inf-full-stack.md:48]: 'Power Infrastructure for deployable architecture' in full-stack variation 'PowerVS workspace' creates in one single deployment Power Virtual Server workspace and connects it with VPC services. Proxy service for public internet access from PowerVS workspace is configured.

File: examples/ibm-catalog/deployable-architectures/extension/README.md [hunk 12]
```diff
@@ -1,66 +1,65 @@
 # IBM Cloud catalog example for Power infrastructure for regulated industries
 
-This example illustrates how to use the `power-infrastructure` module in the IBM catalog. It provisions the following infrastructure:
+This example extends PowerVS infrastructure for deployable architectures in variation PowerVS Workspace incl. Secure Landing Zone (Full Stack solution) with an additional PowerVS Workspace.
```
Comment (Beilmann) [inline examples/ibm-catalog/deployable-architectures/extension/README.md:3]: This example extends an existing PowerVS infrastructure for deployable architectures deployed as full-stack with an additional PowerVS workspace.

File: examples/ibm-catalog/deployable-architectures/full-stack/README.md [hunk 13]
```diff
@@ -0,0 +1,82 @@
+# IBM Cloud catalog example for Power infrastructure for regulated industries
```
Comment (Beilmann) [inline examples/ibm-catalog/deployable-architectures/full-stack/README.md:1]: IBM Cloud catalog example for Power infrastructure for deployable architectures

File: examples/terraform-registry/extension/README.md [hunk 14]
```diff
@@ -1,6 +1,6 @@
 # Standard example for Power infrastructure for regulated industries
```
Comment (Beilmann) [inline examples/terraform-registry/extension/README.md:1]: # Standard example for Power infrastructure for deployable architectures

File: submodules/power_management_services_setup/variables.tf [hunk 15]
```diff
@@ -1,5 +1,5 @@
 variable "access_host_or_ip" {
-  description = "Jump/Bastion server Public IP to reach the target/server_host ip to configure the DNS,NTP,NFS,SQUID services"
+  description = "Jump/Bastion server Public IP to reach the target/server_host ip to configure the DNS,NTP,NFS,SQUID services."
```
Comment (Beilmann) [inline submodules/power_management_services_setup/variables.tf:2]: description = "Jump/Bastion server public IP address to reach the target/server_host ip to configure the DNS,NTP,NFS,SQUID services."

**Review by Beilmann — CHANGES_REQUESTED** [review]

File: reference-architectures/extension/deploy-arch-ibm-pvs-inf-extension.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (SirSpidey) [reply reference-architectures/extension/deploy-arch-ibm-pvs-inf-extension.md:17]: I think that the docs URL is just this one as shown here ending with `/solution-guide`.

File: reference-architectures/extension/deploy-arch-ibm-pvs-inf-extension.md (hunk 9 above)
Comment (SirSpidey) [reply reference-architectures/extension/deploy-arch-ibm-pvs-inf-extension.md:48]: See the suggestion in #358:

> The "extension" variation of the Power Infrastructure for deployable architectures creates a Power Virtual Server workspace and connects it with VPC services. It builds on the VPC service that you created when you deployed Secure infrastructure on VPC for regulated industries or the "full stack" variation of the Power Infrastructure for deployable architecture.

File: reference-architectures/extension/deploy-arch-ibm-pvs-inf-extension.md (hunk 10 above)
Comment (SirSpidey) [reply reference-architectures/extension/deploy-arch-ibm-pvs-inf-extension.md:50]: Had a similar suggestion

File: reference-architectures/full-stack/deploy-arch-ibm-pvs-inf-full-stack.md (hunk 11 above)
Comment (SirSpidey) [reply reference-architectures/full-stack/deploy-arch-ibm-pvs-inf-full-stack.md:48]: Suggest this to coincide with my suggestion for the extension variation. 

The "full stack" variation of the Power Infrastructure for deployable architectures creates a Power Virtual Server workspace and connects it with VPC services. Proxy service for public internet access from PowerVS workspace is configured.

A proxy service for public internet access from the PowerVS workspace is configured. You can optionally configure some management components on VPC (such as an NFS server, NTP forwarder, and DNS forwarder).

File: examples/ibm-catalog/deployable-architectures/extension/README.md [hunk 16]
```diff
@@ -1,66 +1,65 @@
...
+    - Optionally (re)configures the public internet proxy (Squid), DNS forwarder, NTP forwarder and NFS on specified host, and sets the host as server for these services using ansible roles.
 
 ## Before you begin
 
-Install a current version of the [IBM Secure Landing Zone](https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone) module.
+If you do not have an existing PowerVS infrastructure for deployable architectures in variation PowerVS Workspace incl. Secure Landing Zone (Full Stack solution) deployment, create it first.
+
+Alternative: Deploy Secure infrastructure on VPC for regulated industries. Click [here](https://cloud.ibm.com/catalog/content/slz-vpc-with-vsis-a87ed9a5-d130-47a3-980b-5ceb1d4f9280-global#create) to create one and choose the standard flavor. When you deploy this module, use one of supported [configurations](https://github.com/terraform-ibm-modules/terraform-ibm-powervs-infrastructure/tree/main/examples/ibm-catalog/standard-solution/slz_json_configs_for_powervs). Copy the configuration from the link into the `override_json_string` deployment value and set `override` to True.
```
[the remark this answers is not in the document]
Comment (surajsbharadwaj) [reply examples/ibm-catalog/deployable-architectures/extension/README.md:18]: Lets keep this as of now. In future we it changes we will update with the final link

File: examples/terraform-registry/extension/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (surajsbharadwaj) [reply examples/terraform-registry/extension/README.md:3]: Done

File: ibm_catalog.json (hunk 4 above)
Comment (surajsbharadwaj) [reply ibm_catalog.json:8]: Done

File: ibm_catalog.json (hunk 5 above)
Comment (surajsbharadwaj) [reply ibm_catalog.json:199]: Done

File: ibm_catalog.json (hunk 6 above)
Comment (surajsbharadwaj) [reply ibm_catalog.json:193]: Done.

File: ibm_catalog.json (hunk 8 above)
Comment (surajsbharadwaj) [reply ibm_catalog.json:337]: Done

File: examples/ibm-catalog/deployable-architectures/extension/README.md (hunk 12 above)
Comment (surajsbharadwaj) [reply examples/ibm-catalog/deployable-architectures/extension/README.md:3]: Done

**Review by SirSpidey — APPROVED** [review]
👍 Time to get it out there?

**Review by SirSpidey — APPROVED** [review]
👍

File: variables.tf [hunk 17]
```diff
@@ -192,3 +192,14 @@ variable "perform_proxy_client_setup" {
   )
   default = null
 }
+
+variable "example_valid" {
+  description = "To check if the example is valid."
```
Comment (ocofaigh) [inline variables.tf:197]: what does this mean? Needs better descrption

**Review by ocofaigh — CHANGES_REQUESTED** [review]
What is this `example_valid` variable? It's very consuming for consumers. It looks like its not something a consumer should even know about. It should not be exposed as a variable? Could you use this https://stackoverflow.com/a/66682419 ?

File: variables.tf (hunk 17 above)
Comment (surajsbharadwaj) [reply variables.tf]: @ocofaigh https://stackoverflow.com/a/66682419 has been used. example_valid variable removed . Thank you

**Comment by ocofaigh** [comment]
@surajsbharadwaj Please confirm this is going to be a minor version release and has no breaking changes. For the release notes, I think we just need this: `feat: added support for more PowerVS zones: wdc04, us-east, us-south, dal12, dal13, tor01, tok04, osa21, sao01, mon01 (required IBM provider also updated to >=1.49.0)`

**Review by Beilmann — APPROVED** [review]

**Pull request merged**